Lib/test/test_embed.py

```diff
@@ -70,15 +70,18 @@ def setUp(self):
             expecteddir = os.path.join(support.REPO_ROOT, builddir)
         else:
             exepath = os.path.join(builddir, 'Programs')
-            expecteddir = os.path.join(support.REPO_ROOT, 'Programs')
+            expecteddir = os.path.join(builddir, 'Programs')
         self.test_exe = exe = os.path.join(exepath, exename)
         if exepath != expecteddir or not os.path.exists(exe):
             self.skipTest("%r doesn't exist" % exe)
         # This is needed otherwise we get a fatal error:
         # "Py_Initialize: Unable to get the locale encoding
         # LookupError: no codec search functions registered: can't find encoding"
         self.oldcwd = os.getcwd()
-        os.chdir(support.REPO_ROOT)
+        if MS_WINDOWS:
+            os.chdir(support.REPO_ROOT)
+        else:
+            os.chdir(builddir)
```

Comment:
This feels like a bad smell (then again, so are most of the tests in this module...).

The only thing that ought to be affected by the CWD is sys.path, and nothing in the repo root or the build dir should be importable. The only other thing that I can think of is that maybe some of the tests assume that the CWD contains the repo or (more likely, given this change) that it contains the build output.

If it's the latter, there must be some logic in it for Windows to add PCbuild/<plat> back to the CWD to find built files, because they're already out of tree. So we should probably fix that (by changing/removing its assumption about the CWD) so that we aren't multiplexing on OS everywhere.

Reply:
It seems to me this comes from the double meaning of `program_name`.

>Program name used to initialize executable and in early error messages during Python initialization.

https://github.com/python/cpython/blob/5c2e3e828763c8bc4b71429b2e8d61065d7e1a14/Lib/test/test_embed.py#L1158-L1160

When `program_name` above is changed to `python.exe`, release builds pass `test_embed` under `os.chdir(builddir)`.
This matches `program_name` in `getpath.py`. Currently, debug builds gets `python` for the name and pass the test.

Reply:
Debug builds shouldn't pass either, unless that name is actually meaningless (which I think it _mostly_ is on Windows, but apparently not entirely).

What happens if we get that name from `Path(sys.executable).name` instead?

Reply:
>What happens if we get that name from `Path(sys.executable).name` instead?

GitHub CI (debug build) will pass if it is acceptable that getpath.py sets `f"{DEFAULT_PROGRAM_NAME}_d"`(python_d) to `program_name`. To pass in any current directory, `default_program_name()` also needs to check if `Path(sys.executable).name`(python_d.exe) exists.

An alternative is to trim `.exe` and `_d` from actual `config["program_name"]` when comparing in `test_embed`:

https://github.com/python/cpython/blob/ff095e13dfdea64de5c1ad21021ae9b5ca2631f8/Lib/test/test_embed.py#L710-L716

Reply:
> An alternative is to trim .exe and _d from actual config["program_name"] when comparing in test_embed

This seems most reasonable. Maybe we should have explicit tests for getpath ensuring that the name includes `.exe` when needed, though I'd rather put those in the getpath tests than add them into test_embed.

Reply:
@kumaraditya303 Are you ok if I work on the `program_name`? If you already have the fix, please go ahead.

Reply:
Yes, you can go ahead. Thanks